src/librustc_typeck/collect.rs

```diff
@@ -1699,9 +1812,26 @@ fn find_opaque_ty_constraints(tcx: TyCtxt<'_>, def_id: DefId) -> Ty<'_> {
     }
 }
 
+crate fn is_infer_ty(ty: &hir::Ty) -> bool {
```

Comment:
This function feels like it could be missing loads of things. Is it just a best effort function and some things won't have nice diagnostics for that reason? If so, that should be documented

Reply:
This is still not documented

Reply:
Maybe also name it `is_infer_ty_conservative` or sth

Reply:
Also it's not used outside this module, so the `crate` modifier can be removed